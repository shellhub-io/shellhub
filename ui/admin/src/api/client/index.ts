/* tslint:disable */
/* eslint-disable */
/**
 * ShellHub Enterprise OpenAPI
 * > NOTICE: THE API IS NOT STABLE YET; ERROR AND INCONSISTENCIES MAY OCCUR.  ShellHub Enterprise OpenAPI specification.  It documents all routes provided by ShellHub Enterprise. 
 *
 * The version of the OpenAPI document: 0.18.0
 * Contact: contato@ossystems.com.br
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */


export * from "./api";
export * from "./configuration";

