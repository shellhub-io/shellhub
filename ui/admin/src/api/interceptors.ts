/* eslint-disable */
// @ts-nocheck
import { AxiosError, AxiosInstance, AxiosRequestConfig, AxiosResponse } from "axios";
import { store } from "../store";
// import router from "../router/index";
import router from "../router";

const onRequest = (config: AxiosRequestConfig): AxiosRequestConfig => {
  store.dispatch("spinner/setStatus", true);
  return config;
};

const onRequestError = (error: AxiosError): Promise<AxiosError> => {
  store.dispatch("spinner/setStatus", false);
  Promise.reject(error)
};

const onResponse = (response: AxiosResponse): AxiosResponse => {
  store.dispatch("spinner/setStatus", false);
  return response;
};

const onResponseError = async (error: AxiosError): Promise<AxiosError> => {
  store.dispatch("spinner/setStatus", false);
  if (error.response.status === 401) {
    await store.dispatch("auth/logout");
    await router.push({ name: "login" });
  } else if (error.response.status === 402) {
    await router.push({ name: "license" });
  }
  return Promise.reject(error);
};

export function setupInterceptorsTo(axiosInstance: AxiosInstance): AxiosInstance {
  axiosInstance.interceptors.request.use(onRequest, onRequestError);
  axiosInstance.interceptors.response.use(onResponse, onResponseError);
  return axiosInstance;
}
