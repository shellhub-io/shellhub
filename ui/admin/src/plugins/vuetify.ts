// Styles
// import "@mdi/font/css/materialdesignicons.css";
import "vuetify/styles";

// Vuetify
import { createVuetify } from "vuetify";

export default createVuetify({
  theme: {
    defaultTheme: "dark",
  },
});
