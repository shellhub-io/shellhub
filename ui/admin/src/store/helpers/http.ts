import Axios from "axios";
// eslint-disable-next-line @typescript-eslint/ban-ts-comment
// @ts-ignore //* use while route mapping is not in TypeScript
export default () => {
  const axios = Axios.create({
    baseURL: `${window.location.protocol}//${window.location.host}/admin`,
    headers: {
      Authorization: `Bearer ${localStorage.getItem("cloud_token")}`,
    },
  });

  return axios;
};
