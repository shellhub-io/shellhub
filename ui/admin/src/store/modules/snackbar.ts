import { Module } from "vuex";
import { State } from "./../index";

export interface snackbarMessageAndContentType {
  typeMessage: string;
  typeContent: string;
}

export interface SnackbarState {
  snackbarError: boolean;
  snackbarSuccess: boolean;
  snackbarMessageAndContentType: snackbarMessageAndContentType;
  snackbarCopy: boolean;
}

export const snackbar: Module<SnackbarState, State> = {
  namespaced: true,

  state: {
    snackbarError: false,
    snackbarSuccess: false,
    snackbarMessageAndContentType: { typeMessage: "", typeContent: "" },
    snackbarCopy: false,
  },

  getters: {
    snackbarSuccess: (state) => state.snackbarSuccess,
    snackbarError: (state) => state.snackbarError,
    snackbarMessageAndContentType: (state) => state.snackbarMessageAndContentType,
    snackbarCopy: (state) => state.snackbarCopy,
  },

  mutations: {
    setSnackbarSuccessAction: (state, data) => {
      state.snackbarMessageAndContentType = data;
      state.snackbarSuccess = true;
    },

    setSnackbarNoContent: (state, data) => {
      state.snackbarMessageAndContentType = data;
      state.snackbarSuccess = true;
    },

    setSnackbarSuccessDefault: (state) => {
      state.snackbarMessageAndContentType = { typeMessage: "default", typeContent: "" };
      state.snackbarSuccess = true;
    },

    unsetSnackbarSuccess: (state) => {
      state.snackbarSuccess = false;
    },

    setSnackbarErrorLoadingOrAction: (state, data) => {
      state.snackbarMessageAndContentType = data;
      state.snackbarError = true;
    },

    setSnackbarErrorCustom: (state, data) => {
      state.snackbarMessageAndContentType = data;
      state.snackbarError = true;
    },

    setSnackbarErrorDefault: (state) => {
      state.snackbarMessageAndContentType = { typeMessage: "default", typeContent: "" };
      state.snackbarError = true;
    },

    setSnackbarErrorLicense: (state, data) => {
      state.snackbarMessageAndContentType = data;
      state.snackbarError = true;
    },

    unsetSnackbarError: (state) => {
      state.snackbarError = false;
    },

    setSnackbarCopy: (state, value) => {
      state.snackbarMessageAndContentType = { typeMessage: "", typeContent: value };
      state.snackbarCopy = true;
    },

    unsetSnackbarCopy: (state) => {
      state.snackbarCopy = false;
    },
  },

  actions: {
    showSnackbarSuccessAction: ({ commit }, value) => {
      const data = { typeMessage: "action", typeContent: value };
      commit("setSnackbarSuccessAction", data);
    },

    showSnackbarSuccessDefault: ({ commit }) => {
      commit("setSnackbarSuccessDefault");
    },

    unsetShowStatusSnackbarSuccess: ({ commit }) => {
      commit("unsetSnackbarSuccess");
    },

    showSnackbarErrorLoading: ({ commit }, value) => {
      const data = { typeMessage: "loading", typeContent: value };
      commit("setSnackbarErrorLoadingOrAction", data);
    },

    showSnackbarNoContent: ({ commit }) => {
      const data = { typeMessage: "no-content", typeContent: "" };
      commit("setSnackbarNoContent", data);
    },

    showSnackbarErrorAction: ({ commit }, value) => {
      const data = { typeMessage: "action", typeContent: value };
      commit("setSnackbarErrorLoadingOrAction", data);
    },

    showSnackbarErrorCustom: ({ commit }, value) => {
      const data = { typeMessage: "custom", typeContent: value };
      commit("setSnackbarErrorCustom", data);
    },

    showSnackbarErrorLicense: ({ commit }, value) => {
      const data = { typeMessage: "licenseRequired", typeContent: value };
      commit("setSnackbarErrorLicense", data);
    },

    showSnackbarErrorDefault: ({ commit }) => {
      commit("setSnackbarErrorDefault");
    },

    unsetShowStatusSnackbarError: ({ commit }) => {
      commit("unsetSnackbarError");
    },

    showSnackbarCopy: ({ commit }, value) => {
      commit("setSnackbarCopy", value);
    },

    unsetShowStatusSnackbarCopy: ({ commit }) => {
      commit("unsetSnackbarCopy");
    },
  },
};
