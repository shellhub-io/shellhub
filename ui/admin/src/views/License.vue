<template>
  <h1>License Details</h1>
  <v-divider class="mt-2 mb-4" />
  <LicenseComponent />
</template>

<script lang="ts">
import { defineComponent } from "vue";
import LicenseComponent from "../components/License/License.vue";

export default defineComponent({
  components: { LicenseComponent },
});
</script>
