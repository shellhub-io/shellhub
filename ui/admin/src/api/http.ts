import { AxiosInstance } from "axios";
import { setupInterceptorsTo } from "./interceptors";
import * as axiosTs from "./client";
import { Configuration } from "./client";
import { BaseAPI } from "./client/base";

const configuration = new Configuration();
configuration.basePath = `${window.location.protocol}//${window.location.host}`;
configuration.accessToken = localStorage.getItem("cloud_token") || "";

// eslint-disable-next-line import/no-mutable-exports
let adminApi = new axiosTs.AdminApi(configuration);
// eslint-disable-next-line import/no-mutable-exports
let cloudApi = new axiosTs.CloudApi(configuration);

export const createNewClient = () => {
  const newConfiguration = new Configuration();
  newConfiguration.basePath = `${window.location.protocol}//${window.location.host}`;
  newConfiguration.accessToken = localStorage.getItem("cloud_token") || "";
  adminApi = new axiosTs.AdminApi(newConfiguration);
  cloudApi = new axiosTs.CloudApi(newConfiguration);
  return { adminApi, cloudApi };
};

declare module "./client/base" {
  interface BaseAPI {
    getAxios(): AxiosInstance;
    getConfiguration(): Configuration | undefined;
    setConfiguration(configuration: Configuration): void;
  }
}

/** Returns the axios instance */
BaseAPI.prototype.getAxios = function getAxios(this: BaseAPI): AxiosInstance {
  return this.axios;
};

/** Returns the configuration */
// eslint-disable-next-line vue/max-len
BaseAPI.prototype.getConfiguration = function getConfiguration(this: BaseAPI): Configuration | undefined {
  return this.configuration;
};

/** Sets the configuration */
// eslint-disable-next-line vue/max-len
BaseAPI.prototype.setConfiguration = function setConfiguration(this: BaseAPI, configuration: Configuration): void {
  this.configuration = configuration;
};

// eslint-disable-next-line @typescript-eslint/ban-ts-comment
// @ts-ignore
setupInterceptorsTo(adminApi.axios);

export { adminApi, cloudApi };
