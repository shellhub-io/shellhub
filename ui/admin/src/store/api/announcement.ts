import { adminApi } from "./../../api/http";

type Announcement = {
  title: string;
  content: string;
};

const postAnnouncement = async (
  announcement: Announcement,
) => adminApi.createAnnouncement(announcement);

const updateAnnouncement = async (
  uuid: string,
  announcement: Announcement,
) => adminApi.updateAnnouncement(uuid, announcement);

const deleteAnnouncement = async (uuid: string) => adminApi.deleteAnnouncement(uuid);

const getListAnnouncements = async (
  page: number,
  perPage: number,
  orderBy: "asc" | "desc",
) => adminApi.listAnnouncementsAdmin(page, perPage, orderBy);

const getAnnouncement = async (uuid: string) => adminApi.getAnnouncementAdmin(uuid);

export {
  postAnnouncement,
  updateAnnouncement,
  deleteAnnouncement,
  getListAnnouncements,
  getAnnouncement,
};
