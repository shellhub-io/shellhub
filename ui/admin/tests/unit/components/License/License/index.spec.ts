import { createVuetify } from "vuetify";
import { mount, VueWrapper } from "@vue/test-utils";
import { beforeEach, describe, expect, it } from "vitest";
import { createPinia, setActivePinia } from "pinia";
import MockAdapter from "axios-mock-adapter";
import useLicenseStore from "@admin/store/modules/license";
import { adminApi } from "@admin/api/http";
import { SnackbarPlugin } from "@/plugins/snackbar";
import License from "../../../../../src/components/Settings/SettingsLicense.vue";
import routes from "../../../../../src/router";

type LicenseWrapper = VueWrapper<InstanceType<typeof License>>;

const licenseMock = {
  id: "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx",
  expired: false,
  about_to_expire: false,
  grace_period: false,
  issued_at: -1,
  starts_at: -1,
  expires_at: -1,
  allowed_regions: [],
  customer: {
    id: "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx",
    name: "ShellHub",
    email: "contato@ossystems.com.br",
    company: "O.S. Systems",
  },
  features: {
    devices: -1,
    session_recording: true,
    firewall_rules: true,
    billing: false,
    reports: false,
    login_link: false,
  },
};

describe("License", () => {
  let wrapper: LicenseWrapper;
  let mock: MockAdapter;

  beforeEach(async () => {
    setActivePinia(createPinia());
    const vuetify = createVuetify();
    mock = new MockAdapter(adminApi.getAxios());

    const licenseStore = useLicenseStore();
    licenseStore.license = licenseMock;
    mock.onGet("http://localhost:3000/admin/api/license").reply(200);

    wrapper = await mount(License, {
      global: {
        plugins: [vuetify, routes, SnackbarPlugin],
      },
    });
  });

  it("Is a Vue instance", () => {
    expect(wrapper.exists()).toBe(true);
  });

  it("Renders the component", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  Object.keys(licenseMock).forEach((field) => {
    it(`Receives the field ${field} of license state from store`, () => {
      expect(wrapper.vm.license[field]).toEqual(licenseMock[field]);
    });
  });

  it("Renders license fields in template", () => {
    expect(wrapper.find("[data-test=issuedAt-field]").exists()).toBe(true);
    expect(wrapper.find("[data-test=allowedRegions-field]").exists()).toBe(true);
  });

  Object.keys(licenseMock.customer).forEach((customer) => {
    it(`Receives the customer field ${customer} in template`, () => {
      expect(wrapper.find(`[data-test=${customer}]`).exists()).toBe(true);
    });
  });

  Object.keys(licenseMock.features).forEach((feature) => {
    if (["reports", "login_link"].includes(feature)) return;

    it(`Receives the feature ${feature} in template`, () => {
      expect(wrapper.find(`[data-test=${feature}]`).exists()).toBe(true);
    });
  });

  Object.keys(licenseMock.features).forEach((feature) => {
    if (["reports", "login_link"].includes(feature)) return;

    it(`Receives the feature ${feature} with expected text`, () => {
      const value = licenseMock.features[feature];
      const mapFeatureValue = (name: string) => {
        const value = licenseMock.features[name];
        if (typeof value === "boolean") {
          return value ? "mdi-check-circle" : "mdi-close-circle";
        }
        return "unlimited";
      };

      if (typeof value === "boolean") {
        expect(mapFeatureValue(feature)).toBe(value ? "mdi-check-circle" : "mdi-close-circle");
      } else {
        expect(mapFeatureValue(feature)).toBe("unlimited");
      }
    });
  });

  Object.keys(licenseMock.features).forEach((feature) => {
    it(`Receives the feature ${feature} with expected text`, () => {
      const value = licenseMock.features[feature];
      const mapFeatureValue = (name: string) => {
        const value = licenseMock.features[name];
        if (typeof value === "boolean") {
          return value ? "mdi-check-circle" : "mdi-close-circle";
        }
        return "unlimited";
      };

      if (typeof value === "boolean") {
        expect(mapFeatureValue(feature)).toBe(value ? "mdi-check-circle" : "mdi-close-circle");
      } else {
        expect(mapFeatureValue(feature)).toBe("unlimited");
      }
    });
  });
});
