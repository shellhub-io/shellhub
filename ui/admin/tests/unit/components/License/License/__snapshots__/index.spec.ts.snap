// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`License > Renders the component 1`] = `
"<!--v-if-->
<h1 data-v-75f26564="" class="pb-2">License Details</h1>
<div data-v-75f26564="" class="v-card v-theme--light v-card--density-default v-card--variant-elevated w-100 pa-4 bg-background border" data-test="license-card">
  <!---->
  <div class="v-card__loader">
    <div class="v-progress-linear v-theme--light v-locale--is-ltr" style="top: 0px; height: 0px; --v-progress-linear-height: 2px;" role="progressbar" aria-hidden="true" aria-valuemin="0" aria-valuemax="100">
      <!---->
      <div class="v-progress-linear__background"></div>
      <div class="v-progress-linear__buffer" style="width: 0%;"></div>
      <transition-stub name="fade-transition" appear="false" persisted="false" css="true">
        <div class="v-progress-linear__indeterminate">
          <div class="v-progress-linear__indeterminate long"></div>
          <div class="v-progress-linear__indeterminate short"></div>
        </div>
      </transition-stub>
      <!---->
    </div>
  </div>
  <!---->
  <!---->
  <div data-v-75f26564="">
    <div data-v-75f26564="" class="v-card-item pa-0">
      <!---->
      <div class="v-card-item__content">
        <!---->
        <!---->
        <div data-v-75f26564="" class="v-row"><span data-v-75f26564="">Issued at</span><span data-v-75f26564="" class="v-chip v-theme--light v-chip--density-default v-chip--size-default v-chip--variant-tonal" draggable="false" data-test="issued-at-field"><!----><span class="v-chip__underlay"></span>
          <!---->
          <!---->
          <div class="v-chip__content" data-no-activator="">Now</div>
          <!---->
          <!----></span>
        </div>
        <div data-v-75f26564="" class="v-row"><span data-v-75f26564="">Starts at</span><span data-v-75f26564="" class="v-chip v-theme--light v-chip--density-default v-chip--size-default v-chip--variant-tonal" draggable="false" data-test="starts-at-field"><!----><span class="v-chip__underlay"></span>
          <!---->
          <!---->
          <div class="v-chip__content" data-no-activator="">Now</div>
          <!---->
          <!----></span>
        </div>
        <div data-v-75f26564="" class="v-row"><span data-v-75f26564="">Expires at</span><span data-v-75f26564="" class="v-chip v-theme--light v-chip--density-default v-chip--size-default v-chip--variant-tonal" draggable="false" data-test="expires-at-field"><!----><span class="v-chip__underlay"></span>
          <!---->
          <!---->
          <div class="v-chip__content" data-no-activator="">Now</div>
          <!---->
          <!----></span>
        </div>
        <div data-v-75f26564="" class="v-row"><span data-v-75f26564="">Allowed at</span><span data-v-75f26564="" class="v-chip v-theme--light v-chip--density-default v-chip--size-default v-chip--variant-tonal" draggable="false"><!----><span class="v-chip__underlay"></span>
          <!---->
          <!---->
          <div class="v-chip__content" data-no-activator=""><i data-v-75f26564="" class="mdi-earth mdi v-icon notranslate v-theme--light v-icon--size-default mr-2" aria-hidden="true" left=""></i> Global </div>
          <!---->
          <!----></span>
        </div>
      </div>
      <!---->
    </div>
    <hr data-v-75f26564="" class="v-divider v-theme--light my-4" aria-orientation="horizontal" role="separator">
    <div data-v-75f26564="" class="v-card-item pa-0">
      <!---->
      <div class="v-card-item__content">
        <!---->
        <!---->
        <div data-v-75f26564="" class="v-card-title text-h6 text-center">License Owner</div>
        <div data-v-75f26564="" class="v-row" data-test="id"><span data-v-75f26564="">Id</span>
          <div data-v-75f26564=""><span data-v-75f26564="" class="v-chip v-theme--light v-chip--density-default v-chip--size-default v-chip--variant-tonal" draggable="false"><!----><span class="v-chip__underlay"></span>
            <!---->
            <!---->
            <div class="v-chip__content" data-no-activator=""><span data-v-75f26564="" aria-describedby="v-tooltip-v-10" class="hover-text">xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx</span>
              <!--teleport start-->
              <!--teleport end-->
            </div>
            <!---->
            <!----></span>
            <!---->
            <!---->
          </div>
        </div>
        <div data-v-75f26564="" class="v-row" data-test="name"><span data-v-75f26564="">Name</span><span data-v-75f26564="">ShellHub</span></div>
        <div data-v-75f26564="" class="v-row" data-test="email"><span data-v-75f26564="">Email</span><span data-v-75f26564="">contato@ossystems.com.br</span></div>
        <div data-v-75f26564="" class="v-row" data-test="company"><span data-v-75f26564="">Company</span><span data-v-75f26564="">O.S. Systems</span></div>
      </div>
      <!---->
    </div>
    <hr data-v-75f26564="" class="v-divider v-theme--light my-4" aria-orientation="horizontal" role="separator">
    <div data-v-75f26564="" class="v-card-item pa-0">
      <!---->
      <div class="v-card-item__content">
        <!---->
        <!---->
        <div data-v-75f26564="" class="v-card-title text-h6 text-center">Features</div>
        <div data-v-75f26564="" class="v-row" data-test="devices"><span data-v-75f26564="">Devices</span><span data-v-75f26564="" class="v-chip v-theme--light v-chip--density-default v-chip--size-default v-chip--variant-tonal" draggable="false"><!----><span class="v-chip__underlay"></span>
          <!---->
          <!---->
          <div class="v-chip__content" data-no-activator="">Unlimited</div>
          <!---->
          <!----></span>
        </div>
        <div data-v-75f26564="" class="v-row" data-test="session_recording"><span data-v-75f26564="">Session recording</span>
          <div data-v-75f26564=""><i data-v-75f26564="" class="mdi-check-circle mdi v-icon notranslate v-theme--light v-icon--size-default text-success" aria-hidden="true" data-test="included-icon"></i></div>
        </div>
        <div data-v-75f26564="" class="v-row" data-test="firewall_rules"><span data-v-75f26564="">Firewall rules</span>
          <div data-v-75f26564=""><i data-v-75f26564="" class="mdi-check-circle mdi v-icon notranslate v-theme--light v-icon--size-default text-success" aria-hidden="true" data-test="included-icon"></i></div>
        </div>
        <div data-v-75f26564="" class="v-row" data-test="billing"><span data-v-75f26564="">Billing</span>
          <div data-v-75f26564=""><i data-v-75f26564="" class="mdi-close-circle mdi v-icon notranslate v-theme--light v-icon--size-default" style="color: rgb(229, 57, 53); caret-color: #E53935;" aria-hidden="true" data-test="not-included-icon"></i></div>
        </div>
      </div>
      <!---->
    </div>
    <hr data-v-75f26564="" class="v-divider v-theme--light my-4" aria-orientation="horizontal" role="separator">
  </div>
  <div data-v-75f26564="" class="v-card-item">
    <!---->
    <div class="v-card-item__content">
      <!---->
      <!---->
      <div data-v-75f26564="" class="v-card-title text-h6 text-center">License Field</div>
      <div data-v-75f26564="" class="v-input v-input--horizontal v-input--center-affix v-input--density-default v-theme--light v-locale--is-ltr v-file-input mt-4 mb-2">
        <div class="v-input__prepend">
          <!----><i class="mdi-paperclip mdi v-icon notranslate v-theme--light v-icon--size-default v-icon--clickable" role="button" aria-hidden="false" tabindex="0" aria-label="Select license file prepended action"></i>
        </div>
        <div class="v-input__control">
          <div class="v-field v-field--appended v-field--center-affix v-field--variant-outlined v-theme--light v-locale--is-ltr" prependicon="$file">
            <div class="v-field__overlay"></div>
            <div class="v-field__loader">
              <div class="v-progress-linear v-theme--light v-locale--is-ltr" style="top: 0px; height: 0px; --v-progress-linear-height: 2px;" role="progressbar" aria-hidden="true" aria-valuemin="0" aria-valuemax="100">
                <!---->
                <div class="v-progress-linear__background"></div>
                <div class="v-progress-linear__buffer" style="width: 0%;"></div>
                <transition-stub name="fade-transition" appear="false" persisted="false" css="true">
                  <div class="v-progress-linear__indeterminate">
                    <div class="v-progress-linear__indeterminate long"></div>
                    <div class="v-progress-linear__indeterminate short"></div>
                  </div>
                </transition-stub>
                <!---->
              </div>
            </div>
            <!---->
            <div class="v-field__field" data-no-activator="">
              <!----><label class="v-label v-field-label" for="input-v-13">
                <!---->Select license file
              </label><input type="file" accept=".dat" id="input-v-13" aria-describedby="input-v-13-messages">
              <div class="v-field__input">
                <!---->
              </div>
            </div>
            <transition-stub name="expand-x-transition" appear="false" persisted="false" css="true">
              <div class="v-field__clearable" style="display: none;"><i class="mdi-close-circle mdi v-icon notranslate v-theme--light v-icon--size-default v-icon--clickable" role="button" aria-hidden="false" tabindex="-1" aria-label="Clear Select license file"></i></div>
            </transition-stub>
            <div class="v-field__append-inner">
              <!---->
              <!---->
            </div>
            <div class="v-field__outline">
              <div class="v-field__outline__start"></div>
              <div class="v-field__outline__notch"><label class="v-label v-field-label v-field-label--floating" for="input-v-13" aria-hidden="true">
                  <!---->Select license file
                </label></div>
              <div class="v-field__outline__end"></div>
              <!---->
            </div>
          </div>
        </div>
        <!---->
        <div id="input-v-13-messages" class="v-input__details" role="alert" aria-live="polite">
          <transition-group-stub name="slide-y-transition" tag="div" appear="false" persisted="false" css="true" class="v-messages">
            <!---->
          </transition-group-stub>
          <!----><span></span>
          <transition-stub name="slide-y-transition" appear="false" persisted="false" css="true">
            <div class="v-counter" style="display: none;">0 files (0 B in total)</div>
          </transition-stub>
        </div>
      </div><button data-v-75f26564="" type="button" class="v-btn v-btn--disabled v-theme--light v-btn--density-default v-btn--size-default v-btn--variant-outlined" disabled=""><span class="v-btn__overlay"></span><span class="v-btn__underlay"></span>
        <!----><span class="v-btn__content" data-no-activator="">Upload</span>
        <!---->
        <!---->
      </button>
    </div>
    <!---->
  </div>
  <!---->
  <!----><span class="v-card__underlay"></span>
</div>"
`;
